src/vue/directives/tooltip.js

```diff
@@ -0,0 +1,452 @@
+        } else {
+          const freeWindowSpace = (documentWidth - twRect.width) / 2
+          const centeredStartPoint = targetRect.left - freeWindowSpace
+          tooltipWrapper.style.left = -centeredStartPoint + 'px'
+          refreshPositions()
+        }
+      }
+
+      if (twRect.top < 0) {
+        currentPosition = DIRECTIONS.bottom
+        changePosition()
+        refreshPositions()
+      } else if (twRect.bottom > window.innerHeight) {
+        currentPosition = DIRECTIONS.top
+        changePosition()
+        refreshPositions()
+      }
+
+      if (twRect.top < 0 || twRect.bottom > window.innerHeight) {
+        if (twRect.height > window.innerHeight) {
+          tooltipWrapper.style.width = documentWidth + 'px'
+          tooltipWrapper.style.height = 'auto'
+          refreshPositions()
+        } else {
+          const freeWindowSpace = (window.innerHeight - twRect.height) / 2
+          const centeredStartPoint = targetRect.top - freeWindowSpace
+          tooltipWrapper.style.top = -centeredStartPoint + 'px'
+          refreshPositions()
+        }
+      }
+    }
+
+    // add animation depending on current position
+    if (currentPosition === DIRECTIONS.top) {
+      tooltipWrapper.style.transform = `translateY(-${MARGIN_SIZE / 2}px)`
+    } else if (currentPosition === DIRECTIONS.right) {
+      tooltipWrapper.style.transform = `translateX(${MARGIN_SIZE / 2}px)`
+    } else if (currentPosition === DIRECTIONS.bottom) {
+      tooltipWrapper.style.transform = `translateY(${MARGIN_SIZE / 2}px)`
+    } else {
+      tooltipWrapper.style.transform = `translateX(-${MARGIN_SIZE / 2}px)`
+    }
+
+    renderUtilityElements(target)
+  }
+
+  function getPositionStyle (twRect) {
+    const xTargetMidPoint = targetRect.left + (targetRect.width / 2)
+    const yTargetMidPoint = targetRect.top + (targetRect.height / 2)
+    const xTooltipMidPoint = targetRect.left + (twRect.width / 2)
+    return {
+      [DIRECTIONS.top]: {
+        top: targetRect.top - twRect.height - (MARGIN_SIZE / 2),
+        left: targetRect.left - (xTooltipMidPoint - xTargetMidPoint),
+      },
+      [DIRECTIONS.right]: {
+        top: yTargetMidPoint - (twRect.height / 2),
+        left: targetRect.left + targetRect.width + (MARGIN_SIZE / 2),
+      },
+      [DIRECTIONS.left]: {
+        top: yTargetMidPoint - (twRect.height / 2),
+        left: targetRect.left - twRect.width - (MARGIN_SIZE / 2),
+      },
+      [DIRECTIONS.bottom]: {
+        top: targetRect.top + targetRect.height + (MARGIN_SIZE / 2),
+        left: targetRect.left - (xTooltipMidPoint - xTargetMidPoint),
+      },
+    }
+  }
+
+  function renderTooltip (e, binding) {
+    const appContainer = document.getElementById('app')
+    target = e.target ? e.target : e
+    targetRect = target.getBoundingClientRect()
+    position = binding.value.position || DIRECTIONS.TOP
+    currentPosition = position
+    bgColor = binding.value.style.bgColor || COLOR_BLACK
```

Comment:
If any `style` doesn't pass – this part of the code will be broken.
`Cannot read property 'bgColor' of undefined`

Reply:
added safeGet, that will fix current issue